src/poetry/utils/password_manager.py

```diff
@@ -195,18 +195,19 @@ def delete_pypi_token(self, repo_name: str) -> None:
     def get_http_auth(self, repo_name: str) -> dict[str, str | None] | None:
         username = self._config.get(f"http-basic.{repo_name}.username")
         password = self._config.get(f"http-basic.{repo_name}.password")
-        if not username and not password:
+
+        if not username:
             return None
 
         if not password:
             if self.use_keyring:
                 password = self.keyring.get_password(repo_name, username)
-            else:
+            elif not username:
                 return None
```

Comment:
Unhittable, we would have bailed already

Reply:
Not sure that is accurate. When `username="bar"`,  `password=""` and `self.use_keyring == False`, we want to skip to return and not return `None`.

Reply:
when `username="bar"` we will not go into the code branch `elif not username:`

that branch is unhittable because you already made that check and bailed five lines earlier